src/de.rs

```diff
@@ -43,6 +43,21 @@ where
     deserializer.end().map(move |_| t)
 }
 
+/// Same as `from_bytes` but use `limit` as max container depth instead of MAX_CONTAINER_DEPTH`
+pub fn from_bytes_with_limit<'a, T>(bytes: &'a [u8], limit: usize) -> Result<T>
+where
+    T: Deserialize<'a>,
+{
+    if limit > crate::MAX_CONTAINER_DEPTH {
+        return Err(Error::NotSupported(
+            "limit exceeds the max allowed depth 500",
```

Comment:
nit: Perhaps remove " 500" (since we probably don't want to use the crate `const_format`)